modules/network/src/main/java/org/apache/ignite/internal/network/direct/DirectMarshallingUtils.java

```diff
@@ -31,20 +31,31 @@ public class DirectMarshallingUtils {
      * @return Direct message type.
      */
     public static short getShort(ByteBuffer buffer) {
-        byte b0 = buffer.get();
-        byte b1 = buffer.get();
+        int primShift = 0;
+        int prim = 0;
```

Comment:
What is a prim?

Reply:
I'm removing this code, too many problems with it, we'll do it later